x-pack/filebeat/tests/integration/input_reload_test.go

```diff
@@ -0,0 +1,286 @@
+			if time.Now().After(when) {
+				idx = (idx + 1) % len(units)
+				waiting = false
+				return
+			}
+			return
+		}
+		waiting = true
+		when = time.Now().Add(10 * time.Second)
+	}
+	server := &mock.StubServerV2{
+		// The Beat will call the checkin function multiple times:
+		// - At least once at startup
+		// - At every state change (starting, configuring, healthy, etc)
+		// for every Unit.
+		//
+		// Because of that we can't rely on the number of times it is called
+		// we need some sort of state machine to handle when to send the next
+		// policy and when to just re-send the current one.
+		//
+		// If the Elastic-Agent wants the Beat to keep running the same policy,
+		// it will just keep re-sending it every time the Beat calls the checkin
+		// method.
+		CheckinV2Impl: func(observed *proto.CheckinObserved) *proto.CheckinExpected {
+			if management.DoesStateMatch(observed, units[idx], 0) {
+				nextState()
+			}
+			for _, unit := range observed.GetUnits() {
+				if state := unit.GetState(); !(state == proto.State_HEALTHY || state != proto.State_CONFIGURING || state == proto.State_STARTING) {
+					t.Fatalf("Unit '%s' is not healthy, state: %s", unit.GetId(), unit.GetState().String())
+				}
+			}
+			return &proto.CheckinExpected{
+				Units: units[idx],
+			}
+		},
+		ActionImpl: func(response *proto.ActionResponse) error { return nil },
+		// ActionsChan: make(chan *mock.PerformAction, 100),
+	}
+
+	require.NoError(t, server.Start())
+	t.Cleanup(server.Stop)
+
+	filebeat := NewBeat(
+		t,
+		"../../filebeat.test",
+		[]string{
+			"-E", fmt.Sprintf(`management.insecure_grpc_url_for_testing="localhost:%d"`, server.Port),
+			"-E", "management.enabled=true",
+		},
+		tempDir,
+	)
+
+	filebeat.Start()
+
+	require.Eventually(t, func() bool {
+		return filebeat.LogContains("Can only start an input when all related states are finished")
+	}, 5*time.Minute, 100*time.Millisecond,
+		"String 'Can only start an input when all related states are finished' not found on Filebeat logs")
+
+	require.Eventually(t, func() bool {
+		return filebeat.LogContains("file 'flog.log' is not finished, will retry starting the input soon")
+	}, 5*time.Minute, 100*time.Millisecond,
+		"String 'file 'flog.log' is not finished, will retry starting the input soon' not found on Filebeat logs")
+
+	require.Eventually(t, func() bool {
+		return filebeat.LogContains("ForceReload set to TRUE")
+	}, 5*time.Minute, 100*time.Millisecond,
+		"String 'ForceReload set to TRUE' not found on Filebeat logs")
+
+	require.Eventually(t, func() bool {
+		return filebeat.LogContains("Reloading Beats inputs because forceReload is true")
+	}, 5*time.Minute, 100*time.Millisecond,
+		"String 'Reloading Beats inputs because forceReload is true' not found on Filebeat logs")
+
+	require.Eventually(t, func() bool {
+		return filebeat.LogContains("ForceReload set to FALSE")
+	}, 5*time.Minute, 100*time.Millisecond,
+		"String 'ForceReload set to FALSE' not found on Filebeat logs")
```

Comment:
Those are good cases to check the deadline for the tests and set the timeout to something smaller than that. It'd be a lot easier to understand why the test failed if we have the actual error, not the global timeout reached.

Reply:
Do you think it really needs a 5 min timeout?